### PR TITLE
feat(scale-deployment-zero): support any container count

### PR DESCRIPTION
**Pull request by aslafy-z**
## Related Issue(s)

See https://kubernetes.slack.com/archives/CLGR9BJU9/p1671575285828609

## Description

Support multiple containers for the scale-deployment-zero policy.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [ ] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by chipzoller** [comment]
Have you tested to ensure this works properly? If so, can you attach proof manifests?

**Comment by aslafy-z** [comment]
@chipzoller I did not, that's why the PR is still in draft and the checklist item is not checked. I'll send an update once it's done.

**Comment by chipzoller** [comment]
Tests are failing.

File: other/scale_deployment_zero/patchedResource_down.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mydeploy-down
+  annotations:
+    sre.corp.org/troubleshooting-needed: "true"
+spec:
+  replicas: 0
```
Comment (aslafy-z) [inline other/scale_deployment_zero/patchedResource_down.yaml:8]:
```suggestion
```

**Comment by chipzoller** [comment]
Do you still intend to complete this PR?

File: other/scale_deployment_zero/patchedResource_down.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+# TO BE COMPLETED
```
Comment (aslafy-z) [inline other/scale_deployment_zero/patchedResource_down.yaml:9]:
```suggestion
```

**Comment by aslafy-z** [comment]
@chipzoller I have no free time to finish this tests and I'm not using this policy anymore. Would you like to take it over?

**Comment by chipzoller** [comment]
Covering in #636

**Pull request closed without merging**